rebar.lock

```diff
@@ -1,26 +1,38 @@
-{"1.1.0",
-[{<<"certifi">>,{pkg,<<"certifi">>,<<"2.5.1">>},1},
+{"1.2.0",
+[{<<"certifi">>,{pkg,<<"certifi">>,<<"2.5.2">>},1},
  {<<"cf">>,{pkg,<<"cf">>,<<"0.3.1">>},1},
  {<<"erlware_commons">>,{pkg,<<"erlware_commons">>,<<"1.3.1">>},0},
- {<<"hackney">>,{pkg,<<"hackney">>,<<"1.15.2">>},0},
- {<<"idna">>,{pkg,<<"idna">>,<<"6.0.0">>},1},
- {<<"jsx">>,{pkg,<<"jsx">>,<<"2.10.0">>},0},
+ {<<"hackney">>,{pkg,<<"hackney">>,<<"1.16.0">>},0},
+ {<<"idna">>,{pkg,<<"idna">>,<<"6.0.1">>},1},
+ {<<"jsx">>,{pkg,<<"jsx">>,<<"3.0.0">>},0},
  {<<"metrics">>,{pkg,<<"metrics">>,<<"1.0.1">>},1},
  {<<"mimerl">>,{pkg,<<"mimerl">>,<<"1.2.0">>},1},
- {<<"parse_trans">>,{pkg,<<"parse_trans">>,<<"3.3.0">>},2},
- {<<"ssl_verify_fun">>,{pkg,<<"ssl_verify_fun">>,<<"1.1.5">>},1},
- {<<"unicode_util_compat">>,{pkg,<<"unicode_util_compat">>,<<"0.4.1">>},2}]}.
+ {<<"parse_trans">>,{pkg,<<"parse_trans">>,<<"3.3.0">>},1},
+ {<<"ssl_verify_fun">>,{pkg,<<"ssl_verify_fun">>,<<"1.1.6">>},1},
+ {<<"unicode_util_compat">>,{pkg,<<"unicode_util_compat">>,<<"0.5.0">>},2}]}.
 [
 {pkg_hash,[
- {<<"certifi">>, <<"867CE347F7C7D78563450A18A6A28A8090331E77FA02380B4A21962A65D36EE5">>},
+ {<<"certifi">>, <<"B7CFEAE9D2ED395695DD8201C57A2D019C0C43ECAF8B8BCB9320B40D6662F340">>},
  {<<"cf">>, <<"5CB902239476E141EA70A740340233782D363A31EEA8AD37049561542E6CD641">>},
  {<<"erlware_commons">>, <<"0CE192AD69BC6FD0880246D852D0ECE17631E234878011D1586E053641ED4C04">>},
- {<<"hackney">>, <<"07E33C794F8F8964EE86CEBEC1A8ED88DB5070E52E904B8F12209773C1036085">>},
- {<<"idna">>, <<"689C46CBCDF3524C44D5F3DDE8001F364CD7608A99556D8FBD8239A5798D4C10">>},
- {<<"jsx">>, <<"77760560D6AC2B8C51FD4C980E9E19B784016AA70BE354CE746472C33BEB0B1C">>},
+ {<<"hackney">>, <<"5096AC8E823E3A441477B2D187E30DD3FFF1A82991A806B2003845CE72CE2D84">>},
+ {<<"idna">>, <<"1D038FB2E7668CE41FBF681D2C45902E52B3CB9E9C77B55334353B222C2EE50C">>},
+ {<<"jsx">>, <<"20A170ABD4335FC6DB24D5FAD1E5D677C55DADF83D1B20A8A33B5FE159892A39">>},
  {<<"metrics">>, <<"25F094DEA2CDA98213CECC3AEFF09E940299D950904393B2A29D191C346A8486">>},
  {<<"mimerl">>, <<"67E2D3F571088D5CFD3E550C383094B47159F3EEE8FFA08E64106CDF5E981BE3">>},
  {<<"parse_trans">>, <<"09765507A3C7590A784615CFD421D101AEC25098D50B89D7AA1D66646BC571C1">>},
- {<<"ssl_verify_fun">>, <<"6EAF7AD16CB568BB01753DBBD7A95FF8B91C7979482B95F38443FE2C8852A79B">>},
- {<<"unicode_util_compat">>, <<"D869E4C68901DD9531385BB0C8C40444EBF624E60B6962D95952775CAC5E90CD">>}]}
+ {<<"ssl_verify_fun">>, <<"CF344F5692C82D2CD7554F5EC8FD961548D4FD09E7D22F5B62482E5AEAEBD4B0">>},
+ {<<"unicode_util_compat">>, <<"8516502659002CEC19E244EBD90D312183064BE95025A319A6C7E89F4BCCD65B">>}]},
+{pkg_hash_ext,[
```

Comment:
Not sure what this section is. I guess the latest `rebar3` version produces this section in addition?

Reply:
From Fred's comment in the #rebar3 Slack channel:

> pkg_hash is a package-internal hash, pkg_hash_ext is a package-external one that includes metadata. Older versoins of hex only checked the internal one and there was a sneaky attack that could happen when someone used a private hex repo to get a fake package (with the right name) a valid signature for the public hex repo since hex used the same encryption key for both private and public repos

> hex changed their stuff but also moved to an external package hash which prevents testing payloads unless the whole thing (including package name metadata in the index itself) can somehow be imitated (i.e a full mirror with the same keys, which shouldn't happen)

> in an attempt to keep forwards and backwards compatible, since we used the old pkg_hash when it was the only one existing, we amended the lock format to contain both so that new r3 copies at least fetch the safest lock possible, but we didn't instantly explode everybody's build in the ecosystem